File: src/main/java/com/urlshortner/urlshortner/Service/ZooKeeperService.java
package com.urlshortner.urlshortner.Service;

import com.urlshortner.zooKeeper.ZooKeeperClient;
import org.apache.zookeeper.KeeperException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import javax.annotation.PostConstruct;
import java.io.IOException;

@Component
public class ZooKeeperService {

    Logger logger= LoggerFactory.getLogger(getClass());

    @Value("${zookeeper.host}:${zookeeper.port}")
    private String host;
    private ZooKeeperClient client;

    @PostConstruct
    private void job() throws InterruptedException, KeeperException, IOException {
        logger.info("Host: " + host);
        client = new ZooKeeperClient(host, 1000);
        logger.info("\nZooKeeperClient object created\n");
        /*String id = client._zk.create(
                "/counter",
                "1".getBytes(StandardCharsets.UTF_8), ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
        logger.info("\n\n Client id : \n\n" + id);*/

        int range = client.getRangeForClient();
        logger.info("\n\n Range : \n\n" + range);
    }

}

File: src/main/java/com/urlshortner/urlshortner/service/ZooKeeperService.kt
package com.urlshortner.urlshortner.service

import com.urlshortner.urlshortner.model.CounterOperationResult
import com.urlshortner.zooKeeper.ZooKeeperClient
import kotlin.Throws
import java.lang.InterruptedException
import org.apache.zookeeper.KeeperException
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service
import java.io.IOException
import javax.annotation.PostConstruct

@Service
class ZooKeeperService {

    private var logger = LoggerFactory.getLogger(javaClass)

    @Value("\${zookeeper.host}:\${zookeeper.port}")
    private val host: String? = null

    @Value("\${zookeeper.counterDataPath}")
    private val counterDataPath: String? = null

    @Value("\${zookeeper.range.limit}")
    private val limit: Int? = null

    private var client: ZooKeeperClient? = null

    @Autowired
    var counterService: CounterService? = null

    @Throws(InterruptedException::class, KeeperException::class, IOException::class)
    @PostConstruct
    private fun job() {
        logger.info("Host: $host")
        client = ZooKeeperClient(host!!, counterDataPath!!, limit!!)
        logger.info("\nZooKeeperClient object created\n")

        val range = client!!.rangeForClient
        logger.info("\n\n Range : \n\n$range")
        var result = counterService!!.insertCounterRange(range.toLong(), range.toLong()+limit)
        if (result is CounterOperationResult.Failure) {
            logger.info(result.reason)

        } else logger.info("Success")
    }
}
